pkg/registry/batch/job/storage/storage_test.go

```diff
@@ -140,6 +182,178 @@ func TestDelete(t *testing.T) {
+
+func (r *dummyRecorder) AddWarning(agent, text string) {
+	r.agent = agent
+	r.text = text
+	return
+}
+
+func (r *dummyRecorder) getWarning() string {
+	return r.text
+}
+
+var _ warning.Recorder = &dummyRecorder{}
+
+func TestJobDeletion(t *testing.T) {
+	orphanDependents := true
+	orphanDeletionPropagation := metav1.DeletePropagationOrphan
+	backgroundDeletionPropagation := metav1.DeletePropagationBackground
+	job := validNewV1Job()
+	ctx := genericapirequest.NewDefaultContext()
+	key := "/jobs/" + metav1.NamespaceDefault + "/foo"
+	tests := []struct {
+		description   string
+		expectWarning bool
+		deleteOptions *metav1.DeleteOptions
+		requestInfo   *genericapirequest.RequestInfo
+	}{
+		{
+			description:   "DeleteOptions without any policy: warnings expected for batch/v1",
+			expectWarning: true,
+			deleteOptions: &metav1.DeleteOptions{},
+			requestInfo:   &genericapirequest.RequestInfo{APIGroup: "batch", APIVersion: "v1"},
+		},
+		{
+			description:   "DeleteOptions without any policy: no warnings expected for batch/v2",
+			expectWarning: false,
+			deleteOptions: &metav1.DeleteOptions{},
+			requestInfo:   &genericapirequest.RequestInfo{APIGroup: "batch", APIVersion: "v2"},
+		},
+		{
+			description:   "DeleteOptions without any policy: no warnings expected for batch with no APIGroup",
+			expectWarning: false,
+			deleteOptions: &metav1.DeleteOptions{},
+			requestInfo:   &genericapirequest.RequestInfo{APIGroup: "batch", APIVersion: ""},
+		},
+		{
+			description:   "DeleteOptions without orphan: no warnings expected",
+			expectWarning: false,
+			deleteOptions: &metav1.DeleteOptions{OrphanDependents: &orphanDependents},
+			requestInfo:   &genericapirequest.RequestInfo{APIGroup: "batch", APIVersion: "v1"},
+		},
+		{
+			description:   "DeleteOptions with orphan deletion propagation: no warnings expected",
+			expectWarning: false,
+			deleteOptions: &metav1.DeleteOptions{PropagationPolicy: &orphanDeletionPropagation},
+			requestInfo:   &genericapirequest.RequestInfo{APIGroup: "batch", APIVersion: "v1"},
+		},
+		{
+			description:   "DeleteOptions with orphan deletion propagation: no warnings expected",
+			expectWarning: false,
+			deleteOptions: &metav1.DeleteOptions{PropagationPolicy: &backgroundDeletionPropagation},
+			requestInfo:   &genericapirequest.RequestInfo{APIGroup: "batch", APIVersion: "v1"},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.description, func(t *testing.T) {
+			storage, server := newStorage(t)
+			defer server.Terminate(t)
+			defer storage.Job.Store.DestroyFunc()
+			dc := dummyRecorder{agent: "", text: ""}
+			ctx = genericapirequest.WithRequestInfo(ctx, test.requestInfo)
+			ctxWithRecorder := warning.WithWarningRecorder(ctx, &dc)
+			if err := storage.Job.Storage.Create(ctxWithRecorder, key, job, nil, 0, false); err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
+			_, _, err := storage.Job.Delete(ctxWithRecorder, job.Name, rest.ValidateAllObjectFunc, test.deleteOptions)
```

Comment:
you can simplify the tests to exercise both Delete and DeleteCollection in the same testcase, since they should behave identically

Reply:
Done. I was thinking of one test per function but I am fine with it.